derive/src/root_query.rs

```diff
@@ -24,82 +23,501 @@ pub fn root_query_fn(
+    let basic_dependencies = basic_dependencies();
 
+    Ok(quote! {
+        #basic_dependencies
 
-            quote!{
-                pub async fn #name<'a>(
-                    &self,
-                    ctx: &async_graphql::Context<'a>,
-                    filters: Option<#path::Filter>,
-                    pagination: Option<PaginationInput>,
-                    order_by: Option<#path::OrderBy>,
-                ) -> PaginatedResult<#path::Model> {
-                    use sea_orm::prelude::*;
+        #[async_graphql::Object]
+        impl #ident {
+            #(#queries)*
+        }
+    })
+}
+
+pub fn basic_query(name: &Ident, path: &TokenStream) -> TokenStream {
+    quote! {
+        pub async fn #name<'a>(
+            &self,
+            ctx: &async_graphql::Context<'a>,
+            filters: Option<#path::Filter>,
+            pagination: Option<Pagination>,
+            order_by: Option<#path::OrderBy>,
+        ) -> async_graphql::types::connection::Connection<String, #path::Model, ExtraPaginationFields, async_graphql::types::connection::EmptyFields> {
+            use sea_orm::prelude::*;
+            use sea_orm::Iterable;
+            use seaography::itertools::Itertools;
+            use async_graphql::types::connection::CursorType;
+
+            println!("filters: {:?}", filters);
+
+            let db: &crate::DatabaseConnection = ctx.data::<crate::DatabaseConnection>().unwrap();
+            let stmt = #path::Entity::find()
+                .filter(#path::filter_recursive(filters));
+
+            let stmt = #path::order_by(stmt, order_by);
 
-                    let db: &crate::DatabaseConnection = ctx.data::<crate::DatabaseConnection>().unwrap();
-                    let stmt = #path::Entity::find()
-                        .filter(#path::filter_recursive(filters));
+            fn get_result(
+                data: Vec<#path::Model>,
+                has_previous_page: bool,
+                has_next_page: bool,
+                pages: Option<usize>,
+                current: Option<usize>
+            ) -> async_graphql::types::connection::Connection<
+                String,
+                #path::Model,
+                ExtraPaginationFields,
+                async_graphql::types::connection::EmptyFields
+            > {
+                let edges: Vec<async_graphql::types::connection::Edge<String, #path::Model, async_graphql::types::connection::EmptyFields>> = data
+                    .into_iter()
+                    .map(|node| {
+                        let values: Vec<sea_orm::Value> = #path::PrimaryKey::iter()
+                            .map(|variant| {
+                                node.get(variant.into_column())
+                            })
+                            .collect();
 
-                    let stmt = #path::order_by(stmt, order_by);
+                        let cursor_string = CursorValues(values).encode_cursor();
```

Comment:
Hey peeps, not sure if I understand the code here correctly. I think cursor should be constructed from values of sort by columns, not from `PrimaryKey` like here.

For example, when you sort by `name, price` then cursor should be combination of `value of name and value of price`.

Reply:
Hey @phungleson, currently, in seaography, we only support cursor pagination on primary key only.

@karatakis please correct me if I'm get it wrong.